lib/src/package.dart

```diff
@@ -49,31 +49,21 @@ class Package {
   final Pubspec pubspec;
 
   /// The immediate dependencies this package specifies in its pubspec.
-  List<PackageRange> get dependencies => pubspec.dependencies;
+  Map<String, PackageRange> get dependencies => pubspec.dependencies;
 
   /// The immediate dev dependencies this package specifies in its pubspec.
-  List<PackageRange> get devDependencies => pubspec.devDependencies;
+  Map<String, PackageRange> get devDependencies => pubspec.devDependencies;
 
   /// The dependency overrides this package specifies in its pubspec.
-  List<PackageRange> get dependencyOverrides => pubspec.dependencyOverrides;
+  Map<String, PackageRange> get dependencyOverrides => pubspec.dependencyOverrides;
 
   /// All immediate dependencies this package specifies.
   ///
   /// This includes regular, dev dependencies, and overrides.
-  List<PackageRange> get immediateDependencies {
-    var deps = <String, PackageRange>{};
-
-    addToMap(dep) {
-      deps[dep.name] = dep;
-    }
-
-    dependencies.forEach(addToMap);
-    devDependencies.forEach(addToMap);
-
+  Map<String, PackageRange> get immediateDependencies {
+    return {}..addAll(dependencies)..addAll(devDependencies)
     // Make sure to add these last so they replace normal dependencies.
```

Comment:
Putting the comment in the middle of the cascade is kind of weird. How about moving it up and changing "these" -> "overrides"?

Reply:
Done.